core/deployment/src/main/java/io/quarkus/deployment/dev/CompilationProvider.java

```diff
@@ -9,7 +10,7 @@
 import java.util.List;
 import java.util.Set;
 
-public interface CompilationProvider {
+public interface CompilationProvider extends Closeable {
```

Comment:
Maybe we could make make `close` a default method?

Saying that because kogito's implementation (cc @evacchi) would break otherwise

Reply:
IIRC we _extend_ JavaCompilationProvider /cc @mariofusco